src/Chapter1.hs

```diff
@@ -632,7 +641,15 @@ Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
 
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int
+sumLast2 n = uncurry (+) (split n)
+  where
+    split :: Int -> (Int, Int)
+    split n 
+      | n >= 0 =  (mod (div n 10) 10, mod n 10)
+      | mod n 100 == 0 = (0, lastDigit (div n 10))
+      | mod n 10 == 0 = (lastDigit (div n 10), 0)
+      | otherwise = (lastDigit ((div n 10) + 1), lastDigit n)
```

Comment:
That is a wonderful solution! 👏🏼  You correctly noticed that it is the `div` and `mod`, cool 😎 

One hint to make your solution even shorter: you can see that you use both:

```haskell
mod (abs n) 10
div (abs n) 10
```
The standard library has the [`divMod`](https://hackage.haskell.org/package/base-4.14.0.0/docs/Prelude.html#v:divMod) function, that actually combines inside both `div` and `mod`. And this is exactly what you use!.

So you could write it this way:

```haskell
(first, second) = divMod (abs n) 10
```

You can see how we could pattern match on the pair 🙂

Reply:
Combining those two suggestions led to a much shorter, more elegant, and easier to reason through solution! Thanks